packages/arb-token-bridge-ui/src/components/common/Dialog2.tsx

```diff
@@ -114,6 +116,21 @@ export function DialogWrapper(props: DialogProps) {
           isOft={isOftTransfer}
         />
       )
+    case 'import_token':
+      return (
+        <TokenImportDialog
+          {...commonProps}
+          onClose={imported => {
+            if (imported && tokenFromSearchParams) {
+              setSelectedToken(tokenFromSearchParams)
+            } else {
+              setSelectedToken(null)
+            }
+            commonProps.onClose(imported)
+          }}
+          tokenAddress={tokenFromSearchParams!}
```

Comment:
wrap the whole return in `if (tokenFromSearchParams)` instead? then you can remove the `!` and `tokenFromSearchParams` from L124